wcfsetup/install/files/lib/system/template/plugin/IconFunctionTemplatePlugin.class.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace wcf\system\template\plugin;
+
+use wcf\system\template\TemplateEngine;
+
+/**
+ * Template compiler plugin that embeds icons into the page. The
+ * supported sizes are 16, 24, 32, 48, 64, 96, 128 and 144.
+ *
+ * Usage:
+ *  {icon size=16 name='bell'}
+ *
+ * @author Alexander Ebert
+ * @copyright 2001-2022 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package WoltLabSuite\Core\System\Template\Plugin
+ * @since 6.0
+ */
+final class IconFunctionTemplatePlugin implements IFunctionTemplatePlugin
+{
+    private const SIZES = [16, 24, 32, 48, 64, 96, 128, 144];
+
+    private const TYPES = ['brand', 'solid'];
+
+    /**
+     * @inheritDoc
+     */
+    public function execute($tagArgs, TemplateEngine $tplObj)
+    {
+        $size = \intval($tagArgs['size'] ?? 0);
+        $name = $tagArgs['name'] ?? '';
+        $type = $tagArgs['type'] ?? '';
+
+        if (!\in_array($size, self::SIZES)) {
+            throw new \InvalidArgumentException("An unsupported size `{$size}` was requested.");
```

Comment:
```suggestion
            throw new \InvalidArgumentException("An unsupported size '{$size}' was requested.");
```
We usually use single quotes instead of backticks in error messages. Same applies elsewhere.